### PR TITLE
feature: new flags for custom app lock

### PR DESCRIPTION
**Pull request by billypchan**
add `useCustomCodeInsteadOfAccountPassword` for using custom pass code instead of account password.
add `forceCustomAppLock` to choose force using custom app lock or not.

### CODE REVIEW

**Review by marcomaddy1982 — APPROVED** [review]

File: session_manager.json [hunk 1]
```diff
@@ -3,8 +3,14 @@
...
     "blockOnJailbreakOrRoot": false,
     "wipeOnJailbreakOrRoot": false,
+    "forceCustomAppLock": true,
     "forceAppLock": false,
     "appLockTimeout": 10,
     "authenticateAfterReboot": false,
-    "useBiometricsOrAccountPassword": false
+    "useBiometricsOrAccountPassword": true,
+    "useCustomCodeInsteadOfAccountPassword": true,
+    "callCenterConfiguration": {
```
Comment (typfel) [inline session_manager.json:12]: Wasn't this option temporary and it's now deleted @johnxnguyen ?

Comment (johnxnguyen) [reply session_manager.json:12]: Correct

File: session_manager.json [hunk 2]
```diff
@@ -3,8 +3,14 @@
     "blacklistDownloadInterval": 21600,
     "blockOnJailbreakOrRoot": false,
     "wipeOnJailbreakOrRoot": false,
+    "forceCustomAppLock": true,
     "forceAppLock": false,
     "appLockTimeout": 10,
     "authenticateAfterReboot": false,
-    "useBiometricsOrAccountPassword": false
+    "useBiometricsOrAccountPassword": true,
```
Comment (typfel) [inline session_manager.json:10]: Why do we set this to `true`? Won't it have no effect anyway due to `useCustomCodeInsteadOfAccountPassword: true`?

File: session_manager.json (hunk 1 above)
Comment (billypchan) [reply session_manager.json:12]: it is a merge issue and removed, thx!

File: session_manager.json (hunk 2 above)
Comment (typfel) [reply session_manager.json:10]: Bump

Comment (billypchan) [reply session_manager.json:10]: From the requirement, the user always need to unlock with Biometrics by default. The user can switch off in the setting later.

**Review by johnxnguyen — APPROVED** [review]
I find these flags a bit confusing as they seem to conflict with the normal app lock flags. It's unclear at first glance what they mean or how they affect the app, especially when in combination (such as `useBiometricsOrAccountPassword` and `useCustomCodeInsteadOfAccountPassword` (does the latter take priority over the former?)

It would be nice if can just have a single set of flags for App Lock which take both the standard and custom App Lock configurations. Couldn't we do something like:

```
"appLockPasscodeType": "custom" // Can be "custom", "biometric", "account", "device" ,
"forceAppLock": true,
...
```

Or something along those lines? I understand that this would require some changes, so it's more of a proposal for the future, but I'm curious to know what you think.

**Comment by billypchan** [comment]
@johnxnguyen nice suggestion. We can start a new PR for the improvement.

**Pull request merged**